xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -1005,14 +1031,77 @@ private static FilterChainMatch parseFilterChainMatch(
         }
         // TODO(chengyuanzhang): validate if the sum of weights equals to total weight.
         return StructOrError.fromStruct(RouteAction.forWeightedClusters(
-            weightedClusters, hashPolicies, timeoutNano));
+            weightedClusters, hashPolicies, timeoutNano, retryPolicy));
       case CLUSTERSPECIFIER_NOT_SET:
       default:
         return StructOrError.fromError(
             "Unknown cluster specifier: " + proto.getClusterSpecifierCase());
     }
   }
 
+  @Nullable // Return null if we ignore the given policy.
+  private static StructOrError<RetryPolicy> parseRetryPolicy(
+      io.envoyproxy.envoy.config.route.v3.RetryPolicy retryPolicyProto) {
+    int maxAttempts = 2;
+    if (retryPolicyProto.hasNumRetries()) {
+      maxAttempts = retryPolicyProto.getNumRetries().getValue() + 1;
+    }
+    Duration initialBackoff = Durations.fromMillis(25);
+    Duration maxBackoff = Durations.fromMillis(250);
+    if (retryPolicyProto.hasRetryBackOff()) {
+      RetryBackOff retryBackOff = retryPolicyProto.getRetryBackOff();
+      if (!retryBackOff.hasBaseInterval()) {
+        return StructOrError.fromError("No base_interval specified in retry_backoff");
+      }
+      initialBackoff = retryBackOff.getBaseInterval();
+      if (Durations.compare(initialBackoff, Durations.ZERO) <= 0) {
+        return StructOrError.fromError("base_interval in retry_backoff must be positive");
+      }
+      if (Durations.compare(initialBackoff, Durations.fromMillis(1)) < 0) {
+        initialBackoff = Durations.fromMillis(1);
+      }
+      if (retryBackOff.hasMaxInterval()) {
+        maxBackoff = retryPolicyProto.getRetryBackOff().getMaxInterval();
+        if (Durations.compare(maxBackoff, initialBackoff) < 0) {
+          return StructOrError.fromError(
+              "max_interval in retry_backoff cannot be less than base_interval");
```

Comment:
Are we concerned with `initialBackoff < 1ms && maxBackoff < 1ms` making this misleading?

Reply:
Good point. Now consider `maxBackoff < initialBackoff  < 1ms` not valid as well, but `initialBackoff < maxBackoff < 1ms` is valid. PTAL.